### ISSUE TITLE
Automatic Relay  InputType

### ISSUE THREAD

**Issue by RohrerF** [ChilliCream/hotchocolate#773]
**Is your feature request related to a problem? Please describe.**
If I want to write Mutations according to the [relay guide](https://facebook.github.io/relay/graphql/mutations.htm) I have to declare an input type containing all arguments and a ```clientMutationId``` which is returned in the response.
Currently I have to write a lot of annoying Code. One class for the input, one for the payload and in the Mutation itself I have to pass the ```clientMutationId``` from the input to the payload manually.

**Describe the solution you'd like**
I would like to have a very simple and easy solution to speed up development. Maybe something like:

In my mutation/Resolver i would like to have a function as always without the inputType:

```csharp
public User CreateUser(string firstname, string lastname, CancellationToken ctx){
	...
	return new User(....);
}
```

in my `ObjectType` for the Muation I would like to do something like:

```csharp
 descriptor
	.Field(m =>  m.CreateUser(default, default, default))
    .UseRelayInputType()
```

this than should convert this stuff to provide a schema like:

```graphql
type Mutation {
	createUser(input: CreateUserInput!): CreateUserPayload!
}

type CreateUserInput{
	fistname: String
	lastname: String
	clientMutationId: String
}

type CreateUserPayload{
	user: User
	clientMutationId: String
}
```

HC should then automatically pass the `clientMutationId``from the input to the payload and all the fields should be mapped. The GraphQLNonNullType should be respected by this solution

**Comment by michaelstaib** [comment]
I thought about this one on the way back home and I think it is quite easy to implement with the new V9 type system. I think it is about 2 hours of work with testing. If you want to do this one I can show you how to implement this.

**Comment by RohrerF** [comment]
I'm starting to work on that over [here](https://github.com/RohrerF/hotchocolate/tree/automaticInputType)

**Comment by RehanSaeed** [comment]
Any updates on this?

Perhaps if Hot Chocolate included a `RelayInputObjectType<T>` base class with the `clientMutationId` field pre-configured on it, this could be used instead of `InputObjectType<T>`.

```
type Mutation {
	createUser(input: CreateUserInput!): CreateUserPayload!
}

type CreateUserInput{
	fistname: String
	lastname: String
	clientMutationId: String!
}
```

```cs
// Base class for CreateUserInput

public abstract class RelayInputObject<TObjectType> : InputObjectType<TObjectType>
    where TObjectType : InputObjectType
{
    protected override void Configure(IInputObjectTypeDescriptor<TObjectType> descriptor)
    {
        descriptor
            .Field("clientMutationId")
            .Type<NonNullType<StringType>>()
            .Description("The unique identifier of the mutation, sent from the client.");
    }
}
```

Then for the output, a `RelayOutputObjectType<T>` class could look something like this:

```
type RelayOutput{
	value: Foo
	clientMutationId: String
}
```

```cs
public class RelayOutputObject<TObjectType> : ObjectType<TObjectType>
    where TObjectType : ObjectType
{
    protected override void Configure(IObjectTypeDescriptor<TObjectType> descriptor)
    {
        if (descriptor is null)
        {
            throw new ArgumentNullException(nameof(descriptor));
        }

        descriptor.Name("RelayOutput");
        descriptor.Description("The output of a mutation, containing the value and .");

        descriptor
            .Field("value")
            .Type<TObjectType>()
            .Description("The value of the mutation output.");
        descriptor
            .Field("clientMutationId")
            .Type<NonNullType<StringType>>()
            .Description("The unique identifier of the mutation, sent from the client.");
    }
}
```

If Hot Chocolate, knows about both of these types, it could connect the dots and pass the parameter automatically.

**Comment by michaelstaib** [comment]
We are getting ready for an 11 release and this one is now move to the December patch release of 11.

**Comment by RehanSaeed** [comment]
Is this coming in v11.1?

**Comment by tobias-tengler** [comment]
The `clientMutationId` is no longer apart of Relay. It was [removed ](https://github.com/facebook/relay/pull/2349) when Relay Modern was introduced (3 years ago). [Reference](https://github.com/facebook/relay/issues/2077#issuecomment-613625958).
I'm not sure if this should or even needs to be covered by HC at this point!?

**Comment by RohrerF** [comment]
It would still be nice if HC would provide an easy way to convert from "normal" functions like
```csharp
public User CreateUser(string firstname, string lastname, CancellationToken ctx){
	...
	return new User(....);
}
```
to GraphQL like mutations

```
type Mutation {
	createUser(input: CreateUserInput!): CreateUserPayload!
}

type CreateUserInput{
	fistname: String
	lastname: String
}

type CreateUserPayload{
	user: User
} 
```

(not sure if the payload part makes sense in this case but the input certainly would)

**Comment by michaelstaib** [comment]
We postponed it to the June release.

**Comment by PascalSenn** [comment]
This is implemented in 12.4 via [MutationConventions](https://chillicream.com/docs/hotchocolate/defining-a-schema/mutations/#conventions)
